sktime/datatypes/_adapter/polars.py

```diff
@@ -93,17 +144,30 @@ def get_index_names(cols):
                 names[i] = None
         return names
 
+    def set_freq(obj):
+        if isinstance(obj.index, pd.DatetimeIndex):
+            obj.index.freq = pd.infer_freq(obj.index)
+
+        if isinstance(obj.index, pd.MultiIndex):
+            levels = obj.index.levels
+            if isinstance(levels[-1], pd.DatetimeIndex):
+                obj.index.levels[-1].freq = pd.infer_freq(obj.index.levels[-1])
+        return obj
+
     pd_index_names = get_index_names(obj.columns)
 
     if len(pd_index_names) > 0:
         pl_index_names = get_mi_cols(obj)
         obj = obj.set_index(pl_index_names)
         obj.index.names = pd_index_names
+        obj = set_freq(obj)
```

Comment:
could we make this optional, controlled by a parameter? Default false (skip), and use only for the time series like scitypes